appengine/genetics/js/blocks.js

```diff
@@ -0,0 +1,465 @@
+      var name = 'suitor';
+      option.text = Blockly.Msg.VARIABLES_SET_CREATE_GET.replace('%1', name);
+      var xmlField = goog.dom.createDom('field', null, name);
+      xmlField.setAttribute('name', 'VAR');
+      var xmlBlock = goog.dom.createDom('block', null, xmlField);
+      xmlBlock.setAttribute('type', 'variables_get');
+      option.callback = Blockly.ContextMenu.callbackFactory(this, xmlBlock);
+      options.push(option);
+    }
+  }
+};
+
+/**
+ * Defines the JavaScript generation for mateAnswer.
+ * @type {!Function}
+ */
+Blockly.JavaScript['genetics_mateAnswer'] =
+    goog.partial(Blockly.JavaScript['genetics_mouseFunction_'],
+                 'mateAnswer', 'suitor');
+
+Blockly.Blocks['genetics_getSelf'] = {
+  /**
+   * Block for getting the mouse making the decision.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "%1(%2)",
+      "args0": ["getSelf", ""],
+      "output": "Mouse",
+      "colour": Genetics.Blocks.GENETICS_HUE,
+      "tooltip": BlocklyGames.getMsg('Genetics_meTooltip')
+    });
+  }
+};
+
+/**
+ * Defines the JavaScript generation for getSelf.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_getSelf'] = function(block) {
+  // Generate JavaScript for mouse making the decision.
+  return ['getSelf()', Blockly.JavaScript.ORDER_FUNCTION_CALL];
+};
+
+Blockly.Blocks['genetics_getMice'] = {
+  /**
+   * Block for getting all mice in the cage.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "%1(%2)",
+      "args0": ["getMice", ""],
+      "output": "Array",
+      "colour": Genetics.Blocks.GENETICS_HUE,
+      "tooltip": BlocklyGames.getMsg('Genetics_getMiceTooltip')
+    });
+  }
+};
+
+/**
+ * Defines the JavaScript generation for getMice.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_getMice'] = function(block) {
+  // Generate JavaScript for getting mice.
+  return ['getMice()', Blockly.JavaScript.ORDER_FUNCTION_CALL];
+};
+
+/**
+ * Block for getting the properties of a mouse.
+ * @this Blockly.Block
+ */
+Blockly.Blocks['genetics_getProperties'] = {
+  /**
+   * Block for getting mouse properties
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "%1.%2",
+
+      "args0": [
+        {
+          "type": "input_value",
+          "name": "MOUSE",
+          "check": "Mouse"
+        },
+        {
+          "type": "field_dropdown",
+          "name": "PROPERTY",
+          "options": [
+            ['size', 'SIZE'],
+            ['aggressiveness', 'AGGRESSIVENESS'],
+            ['startAggressiveness', 'START_AGGRESSIVENESS'],
+            ['fertility', 'FERTILITY'],
+            ['startingFertility', 'START_FERTILITY'],
+            ['sex', 'SEX'],
+            ['age', 'AGE'],
+            ['id', 'ID'],
+            ['pickFightOwner', 'PICK_FIGHT'],
```

Comment:
Each function (pickFight, chooseMate, mateAnswer) assigned to a mouse is owned by a specific player. So, by "pickFightOwner", I mean the player that wrote the pickFight function that is assigned to the mouse.